core/src/org/pentaho/platform/engine/core/system/objfac/spring/BarrierBeanProcessor.java

```diff
@@ -142,6 +140,16 @@ public void awaitBarrier( String barrierName ){
       // This should never happen
       logger.error( "ServiceBarrier Interrupted", e1 );
     }
-  }  
+  } 
+  
+  /**
+   * 
+   * @param barrierName
+   * @return true if Barrier has no holds, false otherwise
+   */
+  public boolean isAvailable( String barrierName ){
```

Comment:
I don't like saying "yes" to this if the barrier doesn't exist. It can lead to code assuming a barrier is in-place and available when in fact it doesn't even exist